docs/src/design/many_differentials.md

```diff
@@ -0,0 +1,113 @@
+# Design Notes: 1-1, vs many-many relationship between differential types and primal types.
+
+ChainRules (and other earlier packages in Julia AD ecosystem) has a system where one primal type (the type being having its derivative taken) can have multiple possible differential types (the type of the derivative); and that one differential type can corresponds to multiple primal types.
+
+This is in-contrast to the Swift AD efforts.
+Which has one differential type, per primal type.
+(Swift uses the term associated tangent type, rather than differential type).
+
```

Comment:
```suggestion
!!! terminology "(co)vectors" If we wish to be careful with terminology, it should be noted that differentials naturally live in the [_cotangent_ plane](https://en.wikipedia.org/wiki/Cotangent_space) instead of the [tangent plane](https://en.wikipedia.org/wiki/Tangent_space). This distinction is conceptually important, but for convenience, accessibility and consistency with Swift's terminology, we will refer to differentials as vectors in the tangent plane. 

The reason it is conceptually important to distinguish between vectors and covectors is that the notion of a 'dot product' between two vectors relies on defining geometrical notions of distance between two points, whereas the equivalent of a 'dot product' between a vector and a covector does not need any such notion. This is vital for ensuring that tools like gradient descent work whether or not we know how to measure lengths and distances in our space, such as in many machine learning applications. 
```

The second paragraph is less important than the first and can be nixed if it's too wordy for this.

Reply:
`!!!` blocks need their body to be indented with 4 spaced

Reply:
updated